### PR TITLE
[FEATURE] Add dX, dY and residual on GCP Points

### PR DESCRIPTION
**Pull request by fanevanjanahary**
## Description
I added dX, dY and residual on GCP Points file on Georeferencer plugin. Then, I added a checkbox on Transformation Settings for save automatically GCP Points in the raster-modified path.

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

- [ ] Commit messages are descriptive and explain the rationale for changes
- [ ] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [x] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [ ] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and contain sufficient information in the commit message to be documented
- [x] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] New unit tests have been added for core changes
- [x] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTING.md#contributing-to-qgis) before each commit

### CODE REVIEW

File: src/plugins/georeferencer/qgsgeorefplugingui.cpp [hunk 1]
```diff
@@ -1250,7 +1250,7 @@ bool QgsGeorefPluginGui::loadGCPs( /*bool verbose*/ )
 
     QgsPointXY mapCoords( ls.at( 0 ).toDouble(), ls.at( 1 ).toDouble() ); // map x,y
     QgsPointXY pixelCoords( ls.at( 2 ).toDouble(), ls.at( 3 ).toDouble() ); // pixel x,y
-    if ( ls.count() == 5 )
+    if ( ls.count() >= 5 || ls.count() <= 8 )
```
Comment (nyalldawson) [inline src/plugins/georeferencer/qgsgeorefplugingui.cpp]: Shouldn't this be:

```suggestion
    if ( ls.count() >= 5 && ls.count() <= 8 )
```

**Comment by nyalldawson** [comment]
Looking good, this will be a nice addition!

File: src/plugins/georeferencer/qgsgeorefplugingui.cpp (hunk 1 above)
Comment (Fanevanjanahary) [reply src/plugins/georeferencer/qgsgeorefplugingui.cpp]:
```suggestion
    if ( ls.count() == 5 || ls.count() == 8 )
```

**Comment by Fanevanjanahary** [comment]
Thank you for the review code @nyalldawson

**Pull request merged**